pkg/manager/internal/controllers/organizationrole_controller.go

```diff
@@ -0,0 +1,178 @@
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	corev1alpha1 "k8c.io/bulward/pkg/apis/core/v1alpha1"
+	"k8c.io/bulward/pkg/utils/intersection"
+)
+
+type OrganizationRoleReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=bulward.io,resources=organizationroles,verbs=get;list;watch;update
+// +kubebuilder:rbac:groups=bulward.io,resources=organizationroles/status,verbs=get;update;patch
+
+func (r *OrganizationRoleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("organizationRole", req.NamespacedName)
+
+	organizationRole := &corev1alpha1.OrganizationRole{}
+	if err := r.Get(ctx, req.NamespacedName, organizationRole); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+	}
+
+	if !organizationRole.GetDeletionTimestamp().IsZero() {
+		if err := r.handleDeletion(ctx, log, organizationRole); err != nil {
+			return ctrl.Result{}, fmt.Errorf("handling deletion: %w", err)
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if util.AddFinalizer(organizationRole, metav1.FinalizerDeleteDependents) {
+		if err := r.Client.Update(ctx, organizationRole); err != nil {
+			return ctrl.Result{}, fmt.Errorf("updating OrganizationRole finalizers: %w", err)
+		}
+	}
+
+	acceptedRules, err := r.reconcileRole(ctx, organizationRole)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("reconciling Role: %w", err)
+	}
+
+	var isChanged bool
+	if !reflect.DeepEqual(organizationRole.Status.AcceptedRules, acceptedRules) {
+		organizationRole.Status.AcceptedRules = acceptedRules
+		isChanged = true
+	}
+	if !organizationRole.IsReady() {
+		// Update OrganizationRole Status
+		organizationRole.Status.ObservedGeneration = organizationRole.Generation
+		organizationRole.Status.SetCondition(corev1alpha1.OrganizationRoleCondition{
+			Type:    corev1alpha1.OrganizationRoleReady,
+			Status:  corev1alpha1.ConditionTrue,
+			Reason:  "SetupComplete",
+			Message: "OrganizationRole setup is complete.",
+		})
+		isChanged = true
+	}
+	if isChanged {
+		if err := r.Status().Update(ctx, organizationRole); err != nil {
+			return ctrl.Result{}, fmt.Errorf("updating OrganizationRole status: %w", err)
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *OrganizationRoleReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueAllRoles := &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(mapObject handler.MapObject) (out []ctrl.Request) {
+			organizationRoles := &corev1alpha1.OrganizationRoleList{}
```

Comment:
This is going to have scaling issues after a while, and it's wasteful, instead

```suggestion
		ToRequests: handler.ToRequestsFunc(func(mapObject handler.MapObject) (out []ctrl.Request) {
			obj := mapObject.Object.(*corev1alpha1.OrganizationRoleTemplate)
			for _, organizationRole := range obj.Status.Targets {
				out = append(out, ctrl.Request{
					NamespacedName: types.NamespacedName{
						Name:      organizationRole.Name,
						Namespace: mapObject.Meta.GetNamespace(),
					},
				})
			}
			return
```

Reply:
Hmm, this cannot work because the `Targets` in the `Status` are `Organizations` and `Projects`, not `OrganiationRole`.

Reply:
hmmm ok. But you can reconcile only a limited subset for each OrganizationRoleTemplate, and not every OrganizationRole? Because you know which OrganizationRole's are created for each `OrganizationRoleTemplate`

Reply:
Yeah, the reason that I reconcile all OrganizationRole is because `OrganizationRoleTemplate` doens't have a selector, which means `OrganizationRoleTemplate` applies to all Organizations in the system, once we have a selector for `OrganizationRoleTemplate`, we don't need to reconcile every OrganizationRole